ENH: adds optional pairwise beta group significance tests

gregcaporaso opened the pull request:
fixes #50 

[Here's](https://dl.dropboxusercontent.com/u/2868868/temp/unweighted-unifrac-body-site-pairwise-significance.qzv) an example .qzv with the pairwise tests.

I've confirmed that the .qzv looks right when computed with ANOSIM (some of the descriptions of values change), and when it is run without the pairwise tests. 

Pairwise tests are off by default since they can be very slow.

jairideout commented:
travis failures

jairideout commented:
@gregcaporaso, I looked into the travis failures and it's the q2lint issue @jakereps brought up on Slack (too old version of Node on travis). For now let's ignore that error and I'll confirm locally that q2lint passes.

jairideout commented:
Confirmed that q2lint passes locally. @gregcaporaso, can you try adding `dist: trusty` below the `sudo: false` line in .travis.yml?

jairideout commented on q2_diversity/_beta/_visualizer.py:
```diff
@@ -159,15 +172,51 @@ def beta_group_significance(output_dir: str,
+        pairwise_results['q-value'] = multipletests(
+            pairwise_results['p-value'], method='fdr_bh')[1]
+        pairwise_results.sort_index(inplace=True)
+        pairwise_path = os.path.join(output_dir, 'pairwise.csv')
```
It'd also be useful to include a "number of permutations" column in the pairwise results so that the p-values can be interpreted/reported without needing the full visualization and its global test results.

jairideout commented:
I was poking through the example results you posted (BodySite category), and the p-values vs. q-values look a little odd to me:

![screen shot 2017-04-06 at 10 03 32 am](https://cloud.githubusercontent.com/assets/1847232/24766643/f6479a1e-1ab0-11e7-896c-b9ecbca7a4f0.png)

The q-values appear to be the same as the p-values, with some extra precision added. @gregcaporaso can you confirm that these are the expected results? I'm not familiar with statsmodels' `fdr_bh` correction but I would expect the q-values to be noticeably higher since 6 tests were performed.

gregcaporaso commented:
@jairideout, addressed your comments. I computed these q-values with statsmodels directly (independent of this code) and an online calculator, and I got the same values so these should be correct.
